url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/com/wroblicky/andrew/joust/display/DialogBoxDisplay.java

```diff
@@ -0,0 +1,342 @@
+package com.wroblicky.andrew.joust.display;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+import javax.swing.BorderFactory;
+import javax.swing.BoxLayout;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.JSeparator;
+import javax.swing.SwingConstants;
+
+import com.wroblicky.andrew.joust.Util;
+import com.wroblicky.andrew.joust.game.Game;
+import com.wroblicky.andrew.joust.game.PGNViewer;
+import com.wroblicky.andrew.joust.game.board.ChessBoardIterator;
+import com.wroblicky.andrew.joust.game.board.Location;
+import com.wroblicky.andrew.joust.game.chesspiece.ChessPiece;
+import com.wroblicky.andrew.joust.game.move.GameStateChange;
+import com.wroblicky.andrew.joust.game.move.Move;
+import com.wroblicky.andrew.joust.game.move.Turn;
+import com.wroblicky.andrew.joust.pgn.PGNParser;
+
+/**
+ * Manages the ui for displaying a chess game as a dialog box
+ *
+ * @author Andrew Wroblicky
+ *
+ */
+@SuppressWarnings("serial")
+public class DialogBoxDisplay extends JFrame implements ActionListener {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 42 other classes. By comparison, 98% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.